### PR TITLE
Fix the vSphere template uploading error in edge cases

### PR DESCRIPTION
**Pull request by lilirui**
This fixes the issue that when provisioning multiple vSphere VMs from the same template in multiple Go routine, the slower job will be failed because the template is created by faster routine.
This fix just skips the upload if the same template is created or been uploading by other jobs.

@variadico @zquestz @mbhinder @y0ssar1an

### CODE REVIEW

File: virtualmachine/vsphere/util.go [hunk 1]
```diff
@@ -224,7 +224,10 @@ var resetUnitNumbers = func(spec *types.OvfCreateImportSpecResult) {
 var uploadOvf = func(vm *VM, specResult *types.OvfCreateImportSpecResult, lease Lease) error {
 	// Ask the server to wait on the NFC lease
 	leaseInfo, err := lease.Wait()
-	if err != nil {
+	// Stop upload if the template is created already
+	if err != nil && strings.Contains(err.Error(), fmt.Sprintf("The name '%s' already exists", vm.Template)) {
```
Comment (variadico) [inline virtualmachine/vsphere/util.go]: Hey, @mbhinder. Do you think it's safe to make the `Sprintf` less specific. Maybe just "already exists"?

Comment (mbhinder) [inline virtualmachine/vsphere/util.go]: It's probably safer to keep it more specific.

**Comment by zquestz** [comment]
I also agree with @mbhinder I do not believe this code belongs in libretto. This is the callers concern, and we should be bubbling up the error to them, and THEY can choose what to do.

**Comment by lilirui** [comment]
The current code base of Provision function doesn't allow you check the upload error from outside caller, once the error returns, the whole provision process will be terminated. We can lock the mutex base on template name, so that will restrict only one Go routine to do the upload.

**Pull request closed without merging**